git.adoc

```diff
@@ -10,35 +10,33 @@ Sharing some git conveniences `~/.bashrc` aliases and scripts.
 +
 [source,text]
 ----
-#!/usr/bin/bash
+#!/usr/bin/env bash
 
-cd ~/openshift-docs
-if git rev-parse --is-inside-work-tree > /dev/null 2>&1; then
-  echo "Proceeding with rebase."
-else
-  echo "This is not a git repository. Please check your path syntax."
-  exit
+OPENSHIFT_MAIN_BRANCH="main"
+OPENSHIFT_UPSTREAM_REMOTE="upstream"
+OPENSHIFT_PERSONAL_REMOTE="origin"
+
+set -e
+
+if ! git rev-parse --is-inside-work-tree > /dev/null 2>&1; then
+  echo "ERROR: you must execute this script from the OpenShift Git repository." 1>&2
+  exit 1
 fi
 
+git fetch --all
 git stash
 
-CURRENT_BRANCH="$(git branch --show-current)"
+CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
 
-if [ $CURRENT_BRANCH != "main" ]; then
```

Comment:
You don't need to check if you're on main. You can simply store the current branch (even if it's main) and check it out afterwards (even if you're on main).